cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1418 @@
+		byteNumber = []byte(finalNumber)
+		byteName = []byte(currentName)
+		byteName = byteName[:len(byteName)-1]
+		byteName = append(byteName, byteNumber[0])
+		return string(byteName), nil
+	}
+	if s.MatchString(currentName) {
+		stringNumTens := string(currentName[len(currentName)-2])
+		stringNumZeros := string(currentName[len(currentName)-1])
+		tempTens, err := strconv.ParseInt(stringNumTens, 10, 64)
+		if err != nil {
+			return "", err
+		}
+		tempZeros, err := strconv.ParseInt(stringNumZeros, 10, 64)
+		if err != nil {
+			return "", err
+		}
+		tempNumber = tempTens*10 + tempZeros + 1
+		finalNumber = strconv.FormatInt(tempNumber, 10)
+		byteNumber = []byte(finalNumber)
+		byteName = []byte(currentName)
+		byteName = byteName[:len(byteName)-2]
+		byteName = append(byteName, byteNumber[0], byteNumber[1])
+		return string(byteName), nil
+	}
+	return "", errors.New("invalid currentName or pending updates has exceeded 99")
+}
+
+// TrelloUpdateCheckItem updates a check item for trello
+func TrelloUpdateCheckItem(checkItemID, name, state string) error {
+	params := url.Values{}
+	newName, err := NameStateChanges(name, state)
+	if err != nil {
+		return err
+	}
+	params.Set("name", newName)
+	params.Set("state", incomplete)
+	path := fmt.Sprintf(pathUpdateItems, trelloCardID, checkItemID, params.Encode(), usageData.Key, usageData.Token)
+	err = SendAuthReq(http.MethodPut, path, nil)
+	return err
+}
+
+// Update updates the exchange data
+func Update(currentName string, info []ExchangeInfo, updatedInfo ExchangeInfo) []ExchangeInfo {
+	for x := range info {
+		if info[x].Name == currentName {
+			if info[x].CheckType == updatedInfo.CheckType {
+				info[x].Data.GitHubData = updatedInfo.Data.GitHubData
+				info[x].Data.HTMLData = updatedInfo.Data.HTMLData
+				break
+			}
+		}
+	}
+	return info
+}
+
+// UpdateFile updates the given file to match updates.json
+func UpdateFile(name string) error {
+	file, err := json.MarshalIndent(&configData, "", " ")
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(name, file, 0770)
+}
+
+// SendGetReq sends get req
+func SendGetReq(path string, result interface{}) error {
+	var requester *request.Requester
+	if strings.Contains(path, "github") {
+		requester = request.New("Apichecker",
+			common.NewHTTPClientWithTimeout(exchange.DefaultHTTPTimeout),
+			request.NewBasicRateLimit(time.Hour, 60))
+	} else {
+		requester = request.New("Apichecker",
+			common.NewHTTPClientWithTimeout(exchange.DefaultHTTPTimeout),
+			request.NewBasicRateLimit(time.Second, 100))
+	}
+	return requester.SendPayload(&request.Item{
+		Method:  http.MethodGet,
+		Path:    path,
+		Result:  result,
+		Verbose: verbose})
+}
+
+// SendAuthReq sends auth req
+func SendAuthReq(method, path string, result interface{}) error {
+	requester := request.New("Apichecker",
+		common.NewHTTPClientWithTimeout(exchange.DefaultHTTPTimeout),
+		request.NewBasicRateLimit(time.Second*10, 100))
+	return requester.SendPayload(&request.Item{
```

Comment:
the requester is intertwined with the logger in GCT and if you enable verbose, you'll miss out on certain info. It would be a good to generate a default logger(log.SetupGlobalLogger())